ugs-platform/ugs-platform-plugin-joystick/src/main/java/com/willwinder/ugs/nbp/joystick/model/JoystickControl.java

```diff
@@ -0,0 +1,84 @@
+/*
+    Copyright 2020 Will Winder
+
+    This file is part of Universal Gcode Sender (UGS).
+
+    UGS is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    UGS is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with UGS.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.willwinder.ugs.nbp.joystick.model;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The available buttons and axises that can be read by the service.
+ * The controls are divided into two groups:
+ *  - Analog - which will read analog values between 1.0 to -1.0
+ *  - Digital - which will read digital values such as buttons true/false
+ *
+ * @author Joacim Breiler
+ */
+public enum JoystickControl {
```

Comment:
This should support a lot of controllers! One thing I wonder is how it could support controllers with a digital D-Pad?

Reply:
Right now the support is limited. The buttons can't be mapped to a continuous jogging for instance. So if you want to jog it will only allow single stepping.